### PR TITLE
HW-275: v2 optimisations (two datepicker fields)

### PR DESCRIPTION
**Pull request by coldrunKacper**
- Implementing two datepicker input fields to filter Activities by date range instead of a single year/month pair.
- Changing info message copy.

### CODE REVIEW

**Comment by jamienovick** [comment]
Is this using CiviCRM relative dates functionality? @coldrunKacper @mattlinduk

File: CRM/Activityreport/Data.php [hunk 1]
```diff
@@ -20,35 +20,39 @@ class CRM_Activityreport_Data {
...
    */
-  public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0, $startYearMonth = null) {
+  public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0, $startDate = null, $endDate = null) {
     self::$fields = self::getActivityFields();
     self::$emptyRow = self::getEmptyRow();
     self::$multiValues = array();
 
-    $order = empty($startYearMonth) ? 'DESC' : 'ASC';
+    if (empty($startDate)) {
+      $startDate = self::getDefaultStartDate();
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: why not making conditional filtering : 

```php
    $params = array(
      'sequential' => 1,
      'is_current_revision' => 1,
      'is_deleted' => 0,
      'is_test' => 0,
      'return' => implode(',', array_keys(self::$fields)),
      'options' => array(
        'sort' => 'activity_date_time DESC',
        'offset' => $offset,
        'limit' => $limit,
      ),
    );

$activityDateFilter = self::calculateDateFilter($startDate, $endDate);
if (!empty($activityDateFilter)) {
$params['activity_date_time'] = $activityDateFilter;
}

private function prepareAPIDateFilter($startDate, $endDate) {
$apiFilter = null;
if (!empty($startDate) && !empty($endDate)) {
  $apiFilter = array('BETWEEN' => array($startDate, $endDate));
}
else if (!empty($startDate) && empty($endDate)) {
  $apiFilter = array('>=' => $startDate);
} 
else if (empty($startDate) && !empty($endDate)) {
$apiFilter = array('<=' => $endDate);
}

return $apiFilter;
}
```

File: templates/CRM/Activityreport/Page/ActivityReport.tpl [hunk 2]
```diff
@@ -179,103 +186,54 @@
...
             "is_current_revision": 1,
             "is_deleted": 0,
             "is_test": 0,
-            "activity_date_time": {
-              "BETWEEN":[
-                startDateString,
-                endDateString
-              ]
-            }
+            "activity_date_time": {"BETWEEN": [startDate, endDate]}
```
Comment (omarabuhussein) [inline templates/CRM/Activityreport/Page/ActivityReport.tpl]: if you followed my comment above , I think you should follow it here too

File: CRM/Activityreport/Data.php [hunk 3]
```diff
@@ -62,31 +58,26 @@ public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0, $sta
   }
 
   /**
-   * Get datetime of the oldesc Activity or today date if there is no Activity
-   * found (YYYY-MM-DD HH:ii:ss format).
+   * Return an array containing API date filter conditions basing on specified
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: no need to mention that this method return an array or null .. it is already known via @return array|NULL , you just need to describe what this method actually does

File: CRM/Activityreport/Data.php [hunk 4]
```diff
@@ -62,31 +58,26 @@ public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0, $sta
...
+   * dates. Return NULL if dates are not specified.
    *
-   * @return string
+   * @param string $startDate
+   * @param string $endDate
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: please add space between @param and @return

File: CRM/Activityreport/Data.php [hunk 5]
```diff
@@ -62,31 +58,26 @@ public static function get($offset = 0, $limit = 0, $multiValuesOffset = 0, $sta
...
-      'is_deleted' => 0,
-      'is_test' => 0,
-      'return' => 'activity_date_time',
-      'options' => array('sort' => 'activity_date_time ASC', 'limit' => 1),
-    ));
-
-    return !empty($activity['values'][0]['activity_date_time']) ? $activity['values'][0]['activity_date_time'] : date('Y-m-d') . ' 00:00:00';
-  }
+  private static function getAPIDateFilter($startDate, $endDate) {
+    $apiFilter = null;
```
Comment (omarabuhussein) [inline CRM/Activityreport/Data.php]: please add space after this line

**Comment by coldrunKacper** [comment]
@jamienovick No relative dates, now we have two datepicker fields (with absolute dates).

**Pull request merged**